std/conv.d

```diff
@@ -2273,25 +2295,36 @@ to an integral value.
 Params:
     Target = the integral type to convert to
     s = the lvalue of an input range
+    doCount = the flag for deciding to report the number of consumed characters
 
 Returns:
-    A number of type `Target`
+    A number of type `Target` if doCount is set to No.doCount
+    A `tuple` containing a number of type `Target` and a `size_t` if doCount is set to Yes.doCount
 
 Throws:
     A $(LREF ConvException) If an overflow occurred during conversion or
     if no character of the input was meaningfully converted.
 */
-Target parse(Target, Source)(ref Source s)
+auto parse(Target, Source, Flag!"doCount" doCount = No.doCount)(ref Source s)
 if (isSomeChar!(ElementType!Source) &&
     isIntegral!Target && !is(Target == enum))
 {
     static if (Target.sizeof < int.sizeof)
     {
         // smaller types are handled like integers
-        auto v = .parse!(Select!(Target.min < 0, int, uint))(s);
-        auto result = ()@trusted{ return cast(Target) v; }();
-        if (result == v)
-            return result;
+        auto v = .parse!(Select!(Target.min < 0, int, uint), Source, Yes.doCount)(s);
+        auto result = ()@trusted{ return cast(Target) v[0]; }();
```

Comment:
Please the lambda format as `() @trusted { ... }();

Reply:
You could also write it as
`auto result = (() @trusted => cast (Target) v)();`

I think using the `() =>` syntax is prefered to the curly braces `{ ... }`, as you have a oneliner

Reply:
I used the second suggestion. Thank you for the feedback, @edi33416 !